cmd/awscollector/main.go

```diff
@@ -68,50 +66,73 @@ func main() {
 		Version:     version.Version,
 	}
 
+	if err := buildAndParseFlagSet(flagSet, featuregate.GlobalRegistry()); err != nil {
+		logFatal(err)
+	}
+
+	factories, err := defaultcomponents.Components()
+
+	if err != nil {
+		logFatal(fmt.Errorf("failed to build components: %w", err))
+	}
+
 	params := otelcol.CollectorSettings{
 		Factories:      factories,
 		BuildInfo:      info,
 		LoggingOptions: []zap.Option{logger.WrapCoreOpt()},
+		ConfigProvider: config.GetConfigProvider(flagSet),
 	}
 
 	if err = run(params); err != nil {
 		logFatal(err)
 	}
 }
 
+// Parse all the flags manually. We parse the flags manually here so that we can use feature gates when constructing
+// our default component list. Flags also need to be parsed before creating the config provider.
+func buildAndParseFlagSet(fs *flag.FlagSet, featReg *featuregate.Registry) error {
+	fs = config.Flags(featReg)
+
+	if err := flagSet.Parse(os.Args[1:]); err != nil {
+		return err
+	}
+	return nil
+}
+
 func runInteractive(params otelcol.CollectorSettings) error {
 	cmd := newCommand(params)
 	err := cmd.Execute()
 	if err != nil {
 		return fmt.Errorf("application run finished with error: %w", err)
 	}
-
 	return nil
 }
 
 func setCollectorConfigFromExtraCfg(extraCfg *extraconfig.ExtraConfig) {
 	if extraCfg.LoggingLevel != "" {
 		logger.SetLogLevel(extraCfg.LoggingLevel)
 	}
+
 	if extraCfg.AwsProfile != "" {
-		os.Setenv(awsProfileKey, extraCfg.AwsProfile)
+		if err := os.Setenv(awsProfileKey, extraCfg.AwsProfile); err != nil {
+			log.Printf("failed to set env var %s:%v\n", awsProfileKey, err)
+		}
 	}
+
 	if extraCfg.AwsCredentialFile != "" {
-		os.Setenv(awsCredentialFileKey, extraCfg.AwsCredentialFile)
+		if err := os.Setenv(awsCredentialFileKey, extraCfg.AwsCredentialFile); err != nil {
+			log.Printf("failed to set env var %s:%v\n", awsCredentialFileKey, err)
+		}
```

Comment:
These changes are not related to feature gates but I decided to add some additional logging while I was making changes in this function.